File: exercise/kotlin/day13/src/test/kotlin/santamarket/model/FakeCatalog.kt
package santamarket.model

class FakeCatalog : SantamarketCatalog {
    private val products = mutableMapOf<String, Product>()
    private val prices = mutableMapOf<String, Double>()

    override fun addProduct(product: Product, price: Double) {
        products[product.name] = product
        prices[product.name] = price
    }

    override fun getUnitPrice(product: Product): Double {
        return prices[product.name] ?: error("Product not found in catalog")
    }
}

File: exercise/scala/day13/src/test/scala/santamarket/model/FakeCatalog.scala
package santamarket.model

import scala.collection.mutable

class FakeCatalog extends SantamarketCatalog:
  private val products = mutable.Map[String, Product]()
  private val prices = mutable.Map[String, Double]()

  override def addProduct(product: Product, price: Double): Unit =
    products(product.name) = product
    prices(product.name) = price

  override def getUnitPrice(product: Product): Double =
    prices(product.name)